### PR TITLE
Incremental Refactoring

### PR DESCRIPTION
**Pull request by allenlee820202**
1. Prettify server-side codebase.
2. Handle disconnect properly.

@tz70s Please take a look, thanks!

### CODE REVIEW

File: broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
             MqttMessage mqttMessage = (MqttMessage) msg;
             switch (mqttMessage.fixedHeader().messageType()) {
                 case CONNECT:
-                    protocolProcessor.processConnect(ctx.channel(), (MqttConnectMessage) mqttMessage);
                     logger.debug("[MqttMessageHandler] CONNECT");
```
Comment (tz70s) [inline broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java:42]: In my opinion, remove `logger.debug` is better. It is not necessary for debugging here, debug when you need.

Comment (allenlee820202) [reply broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java:42]: I just want to make sure the protocol processor is keep processing the incoming message when debugging this:
In ```MultiClient``` test, the ```Server``` will eventually block in ```MessageDispatcher``` when writing ```PUBLISH``` to the subscriber.
```JAVA
matchSubscription.getSubscriberChannel()
    .writeAndFlush(mqttPublishMessageOut)
    .sync();
```

Maybe this ```logger.debug()``` message can be removed after figuring out this bug.

Comment (allenlee820202) [reply broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java:42]: Of course, this seems to be an unrelated message. It is fine to remove this now.

Comment (tz70s) [reply broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java:42]: Ok, merge this first. Thanks!

**Pull request merged**